### PR TITLE
Sped up Travis build.

### PR DESCRIPTION
**Pull request by mpark**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -292,28 +166,39 @@ before_install:
       export CXX="${CXX}-${VERSION}"
     fi
   - export CXXFLAGS="${CXXFLAGS} -Wall -Wextra -Werror -pedantic-errors"
```
Comment (K-ballo) [inline .travis.yml]: `${CXXFLAGS}` will now be empty here.

File: .travis.yml [hunk 2]
```diff
@@ -292,28 +166,39 @@ before_install:
...
+  - |
+    for CXX_VERSION in "${CXX_VERSIONS[@]}"; do
+      for BUILD_TYPE in "${BUILD_TYPES[@]}"; do
+        BUILD_DIR="build-${CXX_VERSION}-${BUILD_TYPE}"
+        mkdir "${BUILD_DIR}"
+        pushd "${BUILD_DIR}"
+        (
+          set -ex
+          CXXFLAGS="${CXXFLAGS} -std=c++${CXX_VERSION}" cmake \
+              -DCMAKE_BUILD_TYPE=${BUILD_TYPE} ..
```
Comment (K-ballo) [inline .travis.yml]: Instead of splitting the line, just export `CXXFLAGS` on its own line?

File: .travis.yml (hunk 1 above)
Comment (mpark) [reply .travis.yml]: Removed.

File: .travis.yml (hunk 2 above)
Comment (mpark) [reply .travis.yml]: Yep! I was worried about spilling over state to the next loop, but I think it's protected by the parens here.

**Review by K-ballo — APPROVED** [review]

**Pull request merged**